src/components/views/dialogs/InviteDialog.tsx

```diff
@@ -751,7 +866,179 @@ export default class InviteDialog extends React.PureComponent<IInviteDialogProps
         }
     };
 
+    onChangeFilter = (ev: React.MouseEvent<HTMLInputElement>) => {
+        ev.stopPropagation();
+        if (!this.searchIsOnRolePeopleServiceDirectory()) return;
+        const eventTarget = ev.currentTarget;
+        let filterByFavoriteEvent = eventTarget.parentNode.textContent.includes("Favorite");
+        let filterByNameEvent = eventTarget.parentNode.textContent.includes("Name");
+        if(filterByNameEvent) return;
+        if (filterByFavoriteEvent) {
+            let favorite = this.state.favorites;
+            let filteredFavoriteResults;
+            for(let fav in favorite){
+             filteredFavoriteResults = this.state.serverResultsMixin.filter(m => m.user.name.indexOf(fav) !== -1);
+            }
+            this.setState({
+                serverResultsMixin: filteredFavoriteResults,
+                favoriteFilterIsSelected: true,
+                suggestions: []
+            })
+        }
+    }
+
+    _onClearSearchResult = () => {
+        if(!this.searchIsOnRolePeopleServiceDirectory()) return;
+        this.setState({
+            serverResultsMixin: [],
+            numOfRecordsFromSearchAPI: 0
+        });
+    }
+
+    searchIsOnRolePeopleServiceDirectory = () => {
+        if (this.props.kind === KIND_Role_Directory_Search) {
+            return true;
+        } else if (this.props.kind === KIND_People_Directory_Search) {
+            return true;
+        } else if (this.props.kind === KIND_Service_Directory_Search) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * This will be used by service, role and people search by switching api based on search context
+    */
+    _updateFavoritesForCurrentUser() {
+        // Find favorites from relevant api (roles / services)
+        // get user id
+        const user_id_encoded = encodeURI(MatrixClientPeg.get().getUserId());
+        let favorite_api = config.api_base_path + config.prefix + user_id_encoded;
+        if (this.props.kind === KIND_Role_Directory_Search) {
+            favorite_api = config.api_base_path + config.prefix + user_id_encoded
+                + config.search_by_favorite.role_suffix;
+        }
+        fetch(favorite_api, {
+            method: "GET",
+            headers: {
+                'Content-Type': 'application/json',
+            }
+        }).then(res => res.json())
+            .then(response => {
+                this.setState({
+                    favorites: response.favourites
+                })
+            })
```

Comment:
Please handle gracefully if an error occurs e.g.
```
        }).catch(err => { 
            // show the user an error has occured
        })
```

Reply:
Done. Yep, forgot to include catch block there. Thank you for picking this one.